task/sast-snyk-check/0.3/sast-snyk-check.yaml

```diff
@@ -0,0 +1,200 @@
+        
+        # We check if the KFP_GIT_URL variable is set to apply the filters or not
+        if [[ -z "${KFP_GIT_URL}" ]]; then
+          echo "KFP_GIT_URL variable not defined. False positives won't be filtered"
+          mv sast_snyk_check_out_all_findings.json filtered_sast_snyk_check_out.json
+        else
+          echo "Filtering false positives in results files using csfilter-kfp..."
+          
+          CMD=(
+            csfilter-kfp
+            --verbose 
+            --kfp-git-url="${KFP_GIT_URL}"
+          )
+          
+          if [[ -n "${PROJECT_NVR}" ]]; then
+            CMD+=(--project-nvr="${PROJECT_NVR}")
+          fi
+          if [[ -n "${RECORD_EXCLUDED}" ]]; then
+            CMD+=(--record-excluded="${RECORD_EXCLUDED}")
+          fi
+          "${CMD[@]}" sast_snyk_check_out_all_findings.json > filtered_sast_snyk_check_out.json
+          status=$?
+          if [ "$status" -ne 0 ]; then
+            echo "Error: failed to filter known false positives" >&2
+            return 1
+          else
+            echo "Message: Succeed to filter known false positives" >&2
+            SCAN_RESULT="filtered_sast_unicode_check_out.json"
+          fi
+        
+          echo "Results after filtering:"
+          (set -x; csgrep --mode=evtstat filtered_sast_snyk_check_out.json)
+        fi
+
+        csgrep --mode=sarif filtered_sast_snyk_check_out.json > sast_snyk_check_out.sarif
+        
+        if [[ "$SNYK_EXIT_CODE" -eq 0 ]] || [[ "$SNYK_EXIT_CODE" -eq 1 ]]; then
+          TEST_OUTPUT=
+          parse_test_output '$(context.task.name)' sarif sast_snyk_check_out.sarif  || true
+
+        # When the test is skipped, the "SNYK_EXIT_CODE" is 3 and it can also be 3 in some other situation
+        elif [[ "$test_not_skipped" -eq 0 ]]; then
+          note="Task $(context.task.name) success: Snyk code test found zero supported files."
+          ERROR_OUTPUT=$(make_result_json -r SUCCESS -t "$note")
+        else
+          note="Task $(context.task.name) failed: For details, check Tekton task log."
+          ERROR_OUTPUT=$(make_result_json -r ERROR -t "$note")
+        fi
+        echo "${TEST_OUTPUT:-${ERROR_OUTPUT}}" | tee $(results.TEST_OUTPUT.path)
+
+    - name: upload
+      image: quay.io/konflux-ci/oras:latest@sha256:f4b891ee3038a5f13cd92ff4f473faad5601c2434d1c6b9bccdfc134d9d5f820
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      env:
+        - name: IMAGE_URL
+          value: $(params.image-url)
+        - name: IMAGE_DIGEST
+          value: $(params.image-digest)
+      script: |
+        #!/usr/bin/env bash
+
+        UPLOAD_FILE=sast_snyk_check_out.sarif
+        MEDIA_TYPE=application/sarif+json
+
+        if [ -z "${IMAGE_URL}" ] || [ -z "${IMAGE_DIGEST}"  ]; then
+          echo 'No image-url or image-digest param provided. Skipping upload.'
+          exit 0;
+        fi
+
+        UPLOAD_FILES="sast_snyk_check_out.sarif record_excluded.json"
```

Comment:
if the param is a a string and could be specified by user, it could be a filename other than `record_excluded.json`.

Hank had an attempt in his [MR](https://github.com/konflux-ci/build-definitions/pull/1401/files#diff-7e973ab9c156c6c0c394a71e757617d0bc77e24d15dd3b7874dbd61a50cedf24R41) by using boolean `RECORD_EXCLUDED` and use this default value. WDYT?

Reply:
Yes, it should be a boolean value.  If it is set to `false`, the extra file will not be uploaded.

Reply:
Fixed!